lib/Scheduler/aco.cpp

```diff
@@ -169,21 +159,23 @@ SchedInstruction *ACOScheduler::SelectInstruction(std::vector<Choice> ready,
   return ready.back().inst;
 }
 
-InstSchedule *ACOScheduler::FindOneSchedule() {
+std::unique_ptr<InstSchedule> ACOScheduler::FindOneSchedule() {
   SchedInstruction *lastInst = NULL;
-  InstSchedule *schedule = new InstSchedule(machMdl_, dataDepGraph_, true);
+  std::unique_ptr<InstSchedule> schedule(
+      new InstSchedule(machMdl_, dataDepGraph_, true));
```

Comment:
Any reason not to use `llvm::make_unique<InstSchedule>(...)`?

Reply:
The project is currently on llvm 6, and llvm 6 uses c++11.  There is no std::make_unique c++11 (it is introduced in c++14), so we would have to use the identical llvm::make_unique.  In newer versions of llvm that use c++14 they remove llvm::make_unique.  I used the unique_ptr constructor, so we will not have to change this when we eventually update to the newer version of llvm.

Reply:
It's trivial to find and replace `llvm::make_unique` with `std::make_unique` when we're ready to upgrade.

Reply:
Right, the latest LLVM uses std::make_unique and c++14. llvm::make_unique was used with LLVM 6.0.

My understanding was that we use LLVM 6 because of flang. There are multiple projects underway to replace flang e.g. https://github.com/flang-compiler/f18

Reply:
I will change it over to use llvm::make_unique